src/Symfony/Component/Config/Resource/StaticMethodResource.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Resource;
+
+class StaticMethodResource implements SelfCheckingResourceInterface
+{
+    private string $hash;
+
+    /**
+     * @param array $staticMethod A callable represented as [className, methodName]
+     */
+    public function __construct(
+        private $staticMethod,
+    ) {
+    }
+
+    public function isFresh(int $timestamp): bool
+    {
+        $hash = $this->computeHash();
+        $this->hash ??= $hash;
+
+        return $this->hash === $hash;
+    }
+
+    public function __serialize(): array
+    {
+        $this->hash ??= $this->computeHash();
+
+        return [
+            'hash' => $this->hash,
+            'staticMethod' => $this->staticMethod,
+        ];
+    }
+
+    public function __unserialize(array $data): void
+    {
+        $this->hash = array_shift($data);
+        $this->staticMethod = array_shift($data);
+    }
+
+    public function __toString(): string
+    {
+        return 'static_method_resource'.$this->staticMethod[0].'::'.$this->staticMethod[1];
```

Comment:
```suggestion
        return 'static_method_resource:'.$this->staticMethod[0].'::'.$this->staticMethod[1];
```

This will be more readable in case this is investigated, by having a delimiter before the class name.